### ISSUE TITLE
im2col Adreno OpenCL error -54: Invalid worksize

### ISSUE THREAD

**Issue by sivagnanamn** [CNugteren/CLBlast#188]
While trying the new im2col feature on Adreno, facing error -54

```
Channels: 3, height: 448, width: 448, ksize:3, pad: 1, stride:1, offset: 0 --> works fine
Channels: 16, height: 224, width: 224, ksize:3, pad: 1, stride:1, offset: 0 --> works fine
Channels: 32, height: 112, width: 112, ksize:3, pad: 1, stride:1, offset: 0 --> error
CLBlast: OpenCL error: clEnqueueNDRangeKernel: -54
```

**Comment by CNugteren** [comment]
I'm seeing issues on other hardware as well, I'll investigate.

**Comment by CNugteren** [comment]
I've fixed an issue in `6194d43`, perhaps that fixes yours as well. However, [failing automated tests](http://ci.arrayfire.org/view/Other/) on other hardware suggest more issues with `im2col`. I'll investigate more, not sure if I can solve it today, as I don't have direct access to these systems.

**Comment by sivagnanamn** [comment]
@CNugteren Thanks, commit 6194d43 solved my issue. 

I know its an unoptimized version of im2col, here's the im2col metrics for my data dimensions from Adreno 330, hope it helps. Waiting for the optimized version. Cheers.

```
channels;   height;    width;  kernelh;  kernelw;     padh;     padw;  strideh;  stridew;dilationh;dilationw;     offa;     offb;     ms_1; GFLOPS_1;    GBs_1
        3;      448;      448;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;   102.36;      0.0;      0.2
       16;      224;      224;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;   141.05;      0.0;      0.2
       32;      112;      112;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;    75.86;      0.0;      0.2
       64;       56;       56;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;    40.41;      0.0;      0.2
      128;       28;       28;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;    22.51;      0.0;      0.2
      256;       14;       14;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;    12.29;      0.0;      0.2
      512;        7;        7;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;     7.15;      0.0;      0.1
       1K;        7;        7;        3;        3;        1;        1;        1;        1;        1;        1;        0;        0;    12.24;      0.0;      0.2
```
